packages/react/src/components/Tree/TreeTitle.tsx

```diff
@@ -57,15 +57,13 @@ class TreeTitle extends UIComponent<ReactProps<TreeTitleProps>> {
   }
 
   public static defaultProps = {
-    as: 'span',
+    as: 'a',
+    href: '#',
     accessibility: treeTitleBehavior,
   }
 
   protected actionHandlers: AccessibilityActionHandlers = {
-    performClick: e => {
-      e.preventDefault()
-      this.handleClick(e)
-    },
+    performClick: e => this.handleClick(e),
```

Comment:
@jurokapsiar Changes in this PR looks very confusing to me.

```
<Title as='h1' /> => <h1 href='#' />
```

We also will always call `preventDefault` even for other elements.

```
<TreeTitle as={Link} />
```

Reply:
And it actually reverts changes from #1233 as I see.

@priyankar205 please clarify this.

Reply:
We had a long discussion couple of months ago with accessibility testers about the structure of the tree. The challenge is that we cannot fully follow the ARIA pattern as it does not work the way how we need for all screen readers. The structure that was agreed upon was implemented and it was changed in #1233 without sufficient testing. @priyankar205 created this PR to revert and I promised him to take a look at the event propagation.

Reply:
@priyankar205 preventDefault should be called here, inside of performClick